File: jzoffer1-20/18.cpp
/*
struct TreeNode {
	int val;
	struct TreeNode *left;
	struct TreeNode *right;
	TreeNode(int x) :
			val(x), left(NULL), right(NULL) {
	}
};*/
class Solution {
public:
    void Mirror(TreeNode *pRoot) {
        if(! pRoot){
            return;
        }
       TreeNode *p1=NULL;
       p1=pRoot->left;
       pRoot->left=pRoot->right;
       pRoot->right=p1;
       Mirror(pRoot->left);
       Mirror(pRoot->right);
    }
};

File: jzoffer1-20/18.js
/* function TreeNode(x) {
    this.val = x;
    this.left = null;
    this.right = null;
} */
function Mirror(root)
{
    // write code here
    if(!root) return;
    var p1;
    p1=root.left;
    root.left=root.right;
    root.right=p1;
    Mirror(root.left);
    Mirror(root.right);
}